client/src/elm/ItemComment/Update.elm

```diff
@@ -1,63 +1,71 @@
-port module ItemComment.Update
+module ItemComment.Update
     exposing
         ( update
         )
 
-import HttpBuilder exposing (send, toTask, withCredentials, withExpect, withJsonBody, withQueryParams)
-import ItemComment.Decoder exposing (decodeEveryDictListItemComments)
-import ItemComment.Model exposing (EveryDictListItemComments, ItemComment, ItemCommentId, Model, Msg(..))
-import Json.Encode exposing (object, string)
-import RemoteData exposing (..)
-import Utils.WebData exposing (sendWithHandler)
+import Backend.Entities exposing (ItemId)
+import Backend.Item.Model exposing (Item)
+import Backend.Model
+import Backend.Restful exposing (EntityDictList)
+import Editable
+import Editable.WebData
+import EveryDictList exposing (EveryDictList)
+import ItemComment.Model exposing (DelegatedMsg(..), Model, Msg(..))
+import StorageKey exposing (StorageKey)
 
 
-update : Msg -> Model -> ( Model, Cmd Msg, Maybe EveryDictListItemComments )
-update msg model =
+update :
+    Msg
+    -> Model
+    -> ( StorageKey ItemId, { r | items : EntityDictList ItemId Item } )
+    -> ( Model, ( { r | items : EntityDictList ItemId Item }, DelegatedMsg ) )
```

Comment:
Ah, yes, this is the "return an extra message I define, which some caller will know how to handle" pattern.

I'd be tempted to make this a `List DelegatedMsg`, so that it's easy to return more than one ... and, you wouldn't need the `NoOp` then, since you could return an empty list instead.

This is the "gold standard" pattern for decoupling, because the `ItemComment` component (if I may use that word!) is entirely responsible for defining the delegated msg ... the callers then either implement it (typically by translating to their own `Msg` type), or pass it on up to their own caller.

When I'm being lazy, I might just re-use an existing `Msg` type. For instance, instead of the extra return parameter being `List DelegatedMsg`, I might make it `List Backed.Model.Msg` (or possibly some even-more specific existing sg type, if that makes sense). Some caller (if nothing else, `App.Update.update`) will know how to actually perform those messages, so somebody will be able to execute them for me. And, I don't have to define my own "out messages".

But, as I say, defining your own "out messages" is the gold standard, for cases in which you want the very best decoupling. (For instance, in a published package, you'd pretty much have to use the out message standard, or possibly some kind of configuration passed in, since you wouldn't know what the app's `msg` type actually is).

Reply:
Yeah, I should indeed go with a List. Good suggestion 👍 

> This is the "gold standard" pattern for decoupling

Indeed, a big part of this PR was not only to get my hands dirty, but also to come up with rules off-thumbs. My DelegatedMsg is actually wrapping Backend's Msgs -- so we just have a constraint on which Cmds, a specific component (couldn't avoid using that word as-well 😜 ) could trigger.

Reply:
> Yeah, I should indeed go with a List

Actually maybe this is a case for the not yes existing `EverySetList` -- so we can guarantee a Cmd is added only once, but also maintain the order it was added. @rgrempel what do you think?

Reply:
You could get an even stronger constraint by not wrapping an existing `Msg` that someone else definitely implements, but defining your own `OutMsg`, which the caller then must "translate" to its own `Msg` type. Basically, the `OutMsg` would mean: here are some things I need done that I don't know how to do myself -- you, Mr. Caller, must do these things when I ask.

But, that is more wiring & verbosity than I'd usually use, except in cases where it's really important to entirely decouple.

As for `EverySetList`, that certainly could be done. One "gotcha" would be that the things in the `EverySetList` would need to have a `toString` that follows certain rules ... essentially, that it always produces the same result for things that ought to be considered equal, and always produces different results for things that ought not be considered equal. This is true for `EveryDictList` as well, but there we're mostly dealing with simple wrappers around ID's, so there isn't much that can go wrong in our usual cases. A `Msg` is potentially a much more complex type, and may include things that don't deal with `toString` very well (e.g. JSON values). Though, in many cases it may work just fine.